src/index.ts

```diff
@@ -176,3 +176,9 @@ Array.prototype.unionSplit = function (secondArr, comparator, thisArg = this) {
     rightSplit
   };
 };
+
+Array.prototype.distinct = function (func = x => x, thisArg = this) {
+  return thisArg.filter((element, index, arr) => {
+    return arr.findIndex(item => func(element) === func(item)) === index;
```

Comment:
To avoid having too many executions you could declare a scoped const for `func(element)` before checking `arr.findIndex(...)`

so
```typescript

return thisArg.filter((element, index, arr) => {
  const currentCompareVal = func(element); 
  return arr.findIndex(item => currentCompareVal === func(item)) === index;
}
```